analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NonDerivedPrivateClassesShouldBeSealed.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public class Foo
+    {
+
+        private class ClassWontBeExtended // Noncompliant {{Private classes or records which are not derived in the current assembly should be marked as 'sealed'.}}
+//                    ^^^^^^^^^^^^^^^^^^^
+        {
+
+        }
+
+        private sealed class ClassWontBeExtendedButSealed
+        {
+
+        }
+
+        private class PrivateDerivedClass
+        {
+
+        }
+
+        private class PrivateDerivedClassExtension : PrivateDerivedClass // Noncompliant
+        {
+
+        }
+
+        private class PrivateDerivedClassSecondExtension : PrivateDerivedClass
+        {
+
+        }
+
+        private class TheThirdExtension : PrivateDerivedClassSecondExtension // Noncompliant
+        {
+
+        }
+
+        private class PrivateClass // Noncompliant
+        {
+            private class NestedPrivateClass // Noncompliant
+            {
+
+            }
+
+            private class NestedPrivateClassWillBeExtended
+            {
+
+            }
+
+            private class NestedExtension : NestedPrivateClassWillBeExtended // Noncompliant
+            {
+
+            }
+        }
+
+        private abstract class InnerPrivateClass // Compliant
+        {
+
+        }
+
+        private struct AStruct // Compliant, structs cannot be inherited.
+        {
+
+        }
+
+    }
+
```

Comment:
pedantic: we can remove empty lines from all empty methods and class declarations should be delimited by only one empty line.

Reply:
You mean this:
`public class AClass
{
}
`

Reply:
Yes, or even `public class AClass { }` to keep things simple and easy to read.